examples/get_matching/client.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_ext.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_ext.h"
+
+#include "get_me.h"
+
+/*
+ * this file represent a client getting messages from a specified server
+ *
+ * Usage: ./client <server nid> <number of messages we want to get>
+ *
+ */
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_handle_eq_t eqh;
+	ptl_handle_md_t mdh;
+	ptl_event_t ev;
+	long nid;
+	char *p;
+	ptl_md_t md;
+	int get_messages = 0;
+	long wanted_get;
+	char msg[PTL_EV_STR_SIZE];
+	ptl_process_t id_server = { .phys = { .pid = SERVER_PID, .nid = 0 } };
+	char data[MESSAGELENGTH];
+
+	if (argc < 3) {

```

Comment:
We should default to 1 mesage if we only have two arguments I think, instead of failing